Update the XR documentation (#5480)
godotengine/godot-docs: tutorials/vr/openxr/index.rst

.. _doc_openxr_introduction:

OpenXR Plugin
=============

Welcome to the Godot OpenXR documentation!

> 

Reviewer: I think for the official documentation we shouldn't be talking about downloading the source and compiling but purely look at obtaining the plugin from the asset library. We don't want to steer normal users to a version in progress.

Compiling or using the latest unofficial build might be the only thing we should leave either in the readme on github or on the wiki on github

Author: This section doesn't contain instructions for downloading and compiling the plugin. I left them in the github readme instead since there are not relevant here.

The `Getting Started` section however could be improve to link users to the plugin in the assets directory.
Could you provide a blurb for that section.

Reviewer: I'll get something written up this week, slowly getting back into things:)

Introduction
------------

This is the documentation for the `Godot OpenXR plugin <https://github.com/GodotVR/godot_openxr>`__.

> The plugin is supported on Godot 3.4 and up. However it does not support the upcoming Godot 4.0 release.

Reviewer 2: ```suggestion
The plugin is supported on Godot 3.4 and up. However, it does **not** support the upcoming Godot 4.0 release.
```

Getting started
---------------

> To start a new project that supports OpenXR start by opening up the Godot editor and creating a new project. 
> 
> Copy the plugin into this new project in the subfolder ``addons/godot_openxr`` using your favorite file browser. 
> It is very important that it is placed in this exact location in your project folder.
> 
> Back in Godot create a new 3D scene and press the link button to select 
> the ``addons/godot_openxr/scenes/first_person_controller_vr.tscn`` subscene and add it into your scene. 
> Right click on the added node and select ``editable children`` to gain access to some of the components in this subscene:
> 
> .. image:: img/editable_children.png
> 
> **Note:** 
>         With version 1.1.0 and earlier of the plugin, it is important to add a script on the root node and call ``$FPSController.initialise()``.
>         This is no longer required with version 1.1.1 and later.
> 
> This is the bare minimum you need however for good measure we suggest adding a directional light and maybe a few mesh 
> instance nodes so you have something to see. If you add those to the hand nodes you can visualise where your controllers are tracking.
> 
> Your scene should now look something like this:
> 
> .. image:: img/default_scene.png
> 
> Now you can press the run button to start your project and you should be able to look around.

Reviewer 2: ```suggestion
To start a new project that supports OpenXR, start by opening up the Godot editor and creating a new project. 

Copy the plugin into this new project in the subfolder ``addons/godot_openxr/`` using your operating system's file manager. 
It is important that the plugin is placed in this **exact** location in your project folder.

Back in Godot, create a new 3D scene and press the **Instance Child Scene** button
(represented by a chain link icon) in the scene tree dock.
Select the ``addons/godot_openxr/scenes/first_person_controller_vr.tscn`` subscene
and add it to your scene. 
Right-click the added node and select **Editable Children** to gain access
to some of the nodes in this subscene:

.. image:: img/editable_children.png

.. note::

    If using version 1.1.0 or earlier of the plugin, it is important to
    add a script on the root node and call ``$FPSController.initialise()``.
    This is no longer required with version 1.1.1 and later.

This is the bare minimum you need. However, for good measure, we suggest adding
a DirectionalLight node and a few MeshInstance nodes so you have something to see.
If you add those to the hand nodes, you can visualize where your controllers
are tracking.

Your scene should now look something like this:

.. image:: img/default_scene.png

Now you can press the **Run** button in the top-right corner of the editor
to start your project and you should be able to look around.
```

Next steps
----------

> To turn this simple scene into a proper game the sky is the limit. 
> Below there are a few more topics specific to this plugin however the following resources are a good place to continue:
> 
> - `VR starter tutorial <https://docs.godotengine.org/en/stable/tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.html>`__ 
>   in the official documentation may focus on OpenVR but almost everything there applies to OpenXR as well.
> - `Godot XR tools <https://github.com/GodotVR/godot-xr-tools>`__ is a plugin that contains a set of handy sub scene to quickly 
>   implement locomotion, object interaction and UI elements into your XR experience.
> 
> Please check the `Godot Engine community page <https://godotengine.org/community>`__ to find help from other Godot developers. 
> The #XR channel on the Godot Discord has become a vibrant Godot XR community.

Reviewer 2: ```suggestion
To turn this simple scene into a proper game, the sky is the limit. 
Below, there are a few more topics specific to this plugin.
However, the following resources are a good place to continue:

- `VR starter tutorial <https://docs.godotengine.org/en/stable/tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.html>`__
  in the official documentation may focus on OpenVR, but almost everything
  there applies to OpenXR as well.
- `Godot XR tools <https://github.com/GodotVR/godot-xr-tools>`__ is a plugin
  that contains a set of handy sub scene to quickly
  implement locomotion, object interaction and UI elements in your XR experience.

Please check the `Godot Engine community page <https://godotengine.org/community>`__ to find help from other Godot developers. 
The ``#xr`` channel on the Godot Discord has become a vibrant Godot XR community.
```

.. _doc_openxr_introduction:

> OpenXR Plugin
> =============

Reviewer 2: ```suggestion
OpenXR plugin
=============
```
